pkg/controller/elasticsearch/driver/upscale.go

```diff
@@ -100,7 +100,7 @@ func adjustZenConfig(k8sClient k8s.Client, es esv1.Elasticsearch, resources node
 }
 
 func adjustStatefulSetReplicas(
```

Comment:
Can we have a small doc/comment here clarifying why `upscaleState` is a pointer specifically?